algorithms/src/snark/gm17/generator.rs

```diff
@@ -155,6 +155,10 @@ impl<E: PairingEngine> ConstraintSystem<E::Fr> for KeypairAssembly<E> {
     fn num_private_variables(&self) -> usize {
         self.num_private_variables
     }
+
+    fn is_in_setup_mode(&self) -> bool {
+        true
+    }
```

Comment:
Explain: In order to allocate Vec<Boolean> without opening the function, now all the constraint systems must declare whether or not it is in the setup mode.